src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,265 @@
+                            /*using MemoryStream stream = new MemoryStream();*/
+                            UsingDeclare("stream", typeof(MemoryStream), New.Instance(typeof(MemoryStream), Array.Empty<ValueExpression>()), out var stream),
+                            /*content.WriteTo(stream, cancellationToken);*/
+                            Configuration.ApiTypes.GetMultipartFormDataRequestContentWriteToStatment(content, stream, null),
+                            new IfElseStatement(GreaterThan(stream.Property("Position"), new MemberExpression(typeof(int), nameof(int.MaxValue))),
+                            /*return BinaryData.FromStream(stream);*/
+                            Return(BinaryDataExpression.FromStream(stream, false)),
+                            /*return new BinaryData(stream.GetBuffer().AsMemory(0, (int)stream.Position));*/
+                            Return(New.Instance(typeof(BinaryData), new[]{ new InvokeInstanceMethodExpression((new StreamExpression(stream)).GetBuffer, "AsMemory", new[] { Literal(0), new CastExpression(stream.Property("Position"), typeof(int)) }, null, false) }))
+                            ),
+                        };
+        }
+        public static IEnumerable<MethodBodyStatement> BuildToMultipartRequestContentMethodBody(MultipartObjectSerialization? multipart)
+        {
+            return new MethodBodyStatement[]
+                    {
+                        Declare(Configuration.ApiTypes.MultipartRequestContentType, "content", New.Instance(Configuration.ApiTypes.MultipartRequestContentType), out var content),
+                        WriteMultiParts((VariableReference)content, multipart!.Properties).ToArray(),
+                        SerializeAdditionalProperties((VariableReference)content, multipart.AdditionalProperties),
+                        Return(content)
+                    };
+        }
+        private static IEnumerable<MethodBodyStatement> WriteMultiParts(VariableReference multipartContent, IEnumerable<MultipartPropertySerialization> properties)
+        {
+            foreach (MultipartPropertySerialization property in properties)
+            {
+                if (property.SerializedType is { IsNullable: true })
+                {
+                    var checkPropertyIsInitialized = property.SerializedType.IsCollection && !property.SerializedType.IsReadOnlyMemory && property.IsRequired
+                        ? And(NotEqual(property.Value, Null), InvokeOptional.IsCollectionDefined(property.Value))
+                        : NotEqual(property.Value, Null);
+
+                    yield return Serializations.WrapInCheckNotWire(
+                        property,
+                        Literal("MFD"),
+                        InvokeOptional.WrapInIsDefined(
+                            property,
+                    new IfElseStatement(checkPropertyIsInitialized,
+                    WritePropertySerialization(multipartContent, property), null)
+                    ));
```

Comment:
why are we calling the `check not wire` method here?
Do we need to check if this is not wire?
Also please put something for this `MFD`, we should not hard code it everywhere in case it needs to change.